cupy/core/_reduction.pyx

```diff
@@ -577,6 +578,14 @@ cdef class _SimpleReductionKernel(_AbstractReductionKernel):
         else:
             out_type = op.out_types[0]
 
+        # We guessed a routine that requires a C2R casting for the input
+        if (in_args[0].dtype.kind == 'c'
+                and numpy.dtype(op.in_types[0]).kind == 'f'):
```

Comment:
I really hate these two lines, as these are Python attribute accesses, not C. Let me know if there's a better solution...